### PR TITLE
userdb: mangle GECOS field if necessary

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/shared/user-record.c [hunk 1]
```diff
@@ -217,12 +217,21 @@ static int json_dispatch_gecos(const char *name, JsonVariant *variant, JsonDispa
...
-        if (r < 0)
-                return json_log(variant, flags, r, "Failed to allocate string: %m");
+                json_log(variant, flags|JSON_DEBUG, SYNTHETIC_ERRNO(EINVAL), "JSON field '%s' is not a valid GECOS compatible real name, mangling.", strna(name));
+
+                m = mangle_gecos(n);
+                if (!m)
+                        return log_oom();
+
+                free_and_replace(*s, m);
+        } else {
```
Comment (keszybz) [inline src/shared/user-record.c]: Hmm, the order is "valid-invalid" above "invalid-valid" here. My OCD is triggered ;)

Comment (poettering) [reply src/shared/user-record.c]: Now you triggered my OCD, too.

Force pushed a new version. only change: the two branches are swapped. No other changes.

**Comment by keszybz** [comment]
```
 ../src/shared/user-record.c: In function ‘json_dispatch_gecos’:
../src/shared/user-record.c:209:13: error: unused variable ‘r’ [-Werror=unused-variable]
  209 |         int r;
      |             ^
cc1: all warnings being treated as errors
```

**Comment by poettering** [comment]
ouch. dropped the `r`

**Pull request merged**